kornia/geometry/epipolar/essential.py

```diff
@@ -19,6 +22,181 @@
     "relative_camera_motion",
 ]
 
+# Reference
+# Nistér, David. An efficient solution to the five-point relative pose problem. 2004.
+# https://github.com/danini/graph-cut-ransac/blob/master/src/pygcransac/include/estimators/solver_essential_matrix_five_point_nister.h
+# Wei T, Patel Y, Matas J, Barath D. Generalized differentiable RANSAC[J]. arXiv preprint arXiv:2212.13185, 2023.
+# https://github.com/weitong8591/differentiable_ransac/blob/main/estimators/essential_matrix_estimator_nister.py
+
+
+def run_5point(points1: torch.Tensor, points2: torch.Tensor, weights: Optional[torch.Tensor] = None) -> torch.Tensor:
+    r"""Compute the essential matrix using the 5-point algorithm from Nister.
+
+    The linear system is solved by Nister's 5-point algorithm.
+
+    Args:
+        points1: A set of carlibrated points in the first image with a tensor shape :math:`(B, N, 2), N>=8`.
+        points2: A set of points in the second image with a tensor shape :math:`(B, N, 2), N>=8`.
+        weights: Tensor containing the weights per point correspondence with a shape of :math:`(B, N)`.
+
+    Returns:
+        the computed fundamental matrix with shape :math:`(B, 3, 3)`.
+    """
+    KORNIA_CHECK_SHAPE(points1, ['B', 'N', '2'])
+    KORNIA_CHECK_SAME_SHAPE(points1, points2)
+    KORNIA_CHECK(points1.shape[1] >= 5, "Number of points should be >=5")
+    if weights is not None:
+        KORNIA_CHECK_SAME_SHAPE(points1[:, :, 0], weights)
+
+    batch_size, _, _ = points1.shape
+    x1, y1 = torch.chunk(points1, dim=-1, chunks=2)  # Bx1xN
+    x2, y2 = torch.chunk(points2, dim=-1, chunks=2)  # Bx1xN
+    ones = torch.ones_like(x1)
+
+    # build equations system and solve DLT
```

Comment:
@ducha-aiki i've seen this code many times, maybe as post task we could have this into solver a `solve_dlt` ?

Reply:
Hey! Thanks, I borrowed a wrong comment from 8PC about DLT on 5PC, it is incorrect. I change it, and I think 5PC should be here ad there is 7PC inside 'fundamental.py'. As we have minimal solvers 5PC and 7PC for E and F estimation, we can put the non-minimal solver, 8PC to solve_DLT.